fastjet/internal/delaunay/delaunay.go

```diff
@@ -7,27 +7,355 @@ package delaunay
 import (
 	"fmt"
 	"math/rand"
+
+	"go-hep.org/x/hep/fastjet/internal/predicates"
 )
 
+// Delaunay holds necessary information for the delaunay triangulation.
 type Delaunay struct {
+	// triangles is a slice of all triangles that have been created. It is used to get the final
+	// list of triangles in the delaunay triangulation.
+	triangles triangles
+	// root is a triangle that contains all points. It is used as the starting point in the hierarchy
+	// to locate a point.
+	root            *Triangle
+	useHierarchical bool // useHierarchical indicates which method to use to locate a point.
+	r               *rand.Rand
 }
 
+// HierarchicalDelaunay creates a Delaunay Triangulation using the delaunay hierarchy.
+//
+// The worst case time complexity is O(n*log(n)).
+//
+// The three root points (-2^100,-2^100), (2^100,-2^100) and (0,2^100) can't be in the circumcircle of any three non-collinear points in the
```

Comment:
Can't you modify the algorithm so that the `root` triangle is a virtual one that contains any point, that is, the location of its vertices is irrelevant?